spec/engine/miq_ae_method_service/miq_ae_service_model_base_spec.rb

```diff
@@ -94,6 +94,28 @@
     end
   end
 
+  describe '.to_yaml' do
+    let(:service)     { FactoryBot.create(:service, :name => 'test_service') }
+    let(:svc_service) { MiqAeMethodService::MiqAeServiceService.find(service.id) }
+
+    it 'exports only class name and ID' do
+      expect(svc_service.to_yaml)
+        .to eq("--- !ruby/object:#{svc_service.class.name}\nid: #{svc_service.id}\n")
+    end
+
+    it 'loads object from yaml' do
+      expect(YAML.safe_load(svc_service.to_yaml)).to eq(svc_service)
+    end
+
+    it 'returns nil for objects without related active record instance' do
+      yaml = svc_service.to_yaml
+      service.delete
+      # This test and tested behaviour is in-progress state. Right now we are
+      # getting back the invalid service_model instance with blank @object variable.
+      expect(YAML.safe_load(yaml)).to eq(nil)
```

Comment:
@pkomanek I do not think we will be able to accomplish this which means this approach is not going to be the right solution.  In discussing this issue with @mkanoor and @tinaafitz we talked about going in a different direction to solve the original problem.

Suggesting we close this PR for now and I will update the git issue with more details.